infra/general/languageCodeManager.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+final class languageCodeManager
+{
+    private static $arrayISO639_1 = null;
+    private static $arrayISO639_T = null;
+    private static $arrayKalturaName = null;
+
+    const ISO639_1 = 0; //lowercase
+    const ISO639_T = 1;
+    const ISO639_B = 2;
+    const ISO_NAME = 3;
+    const ISO_NATIVE_NAME = 4;
+    const KALTURA_NAME = 5;
+
+    public function __construct()
+    {
+        $cacheFileName = kConf::get("cache_root_path") . "/infra/languages_manager_cache.php";
+        if(file_exists($cacheFileName))

```

Comment:
Before including, verify that the arrays were not loaded already.
write logic of isAlreadyLoaded in an external static function.